File: mavkit-python/examples/tlog_parse.py
"""Parse a TLOG file and print all entries."""

import asyncio
import json
import sys

import mavkit


async def main():
    if len(sys.argv) < 2:
        print("usage: tlog_parse.py <path.tlog>")
        sys.exit(1)

    path = sys.argv[1]
    tlog = await mavkit.TlogFile.open(path)

    # Print time range
    time_range = await tlog.time_range()
    if time_range is not None:
        first, last = time_range
        print(f"time range: {first} .. {last} usec")
    else:
        print("empty file")
        return

    # Print all entries
    entries = await tlog.entries()
    for entry in entries:
        fields = json.loads(entry.message_json())
        print(f"[{entry.timestamp_usec}] {entry.message_name} (id={entry.message_id})")
        print(f"  {fields}")

    print(f"{len(entries)} entries total")


asyncio.run(main())

File: examples/tlog_parse.rs
use mavkit::tlog::TlogFile;
use mavlink::Message;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let path = std::env::args()
        .nth(1)
        .expect("usage: tlog_parse <path.tlog>");

    let tlog = TlogFile::open(&path).await?;

    // Print time range
    if let Some((first, last)) = tlog.time_range().await? {
        println!("time range: {first} .. {last} usec");
    } else {
        println!("empty file");
        return Ok(());
    }

    // Print all entries
    let mut reader = tlog.entries().await?;
    let mut count = 0u64;
    while let Some(entry) = reader.next().await? {
        println!(
            "[{}] {} (id={})",
            entry.timestamp_usec,
            entry.message.message_name(),
            entry.message.message_id(),
        );
        count += 1;
    }
    println!("{count} entries total");

    Ok(())
}
